flake.nix

```diff
@@ -0,0 +1,123 @@
+{
+  description =
+    "tinygrad: For something between PyTorch and karpathy/micrograd. Maintained by tiny corp.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils }@I:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        P = import nixpkgs {
+          inherit system;
+          config.allowUnfree = true;
+        };
+        B = builtins;
+        L = P.lib;
+
+        pythonVersion = "310";
+        pythonPackageStr = "python${pythonVersion}";
+        pythonNixpkgs = P."${pythonPackageStr}Packages";
+
+        tinygrad = pythonNixpkgs.buildPythonPackage {
+          pname = "tinygrad";
+          version = "v.0.6.0";
+          src = ./.;
+
+          propagatedBuildInputs = with pythonNixpkgs; [
+            networkx
+            numpy
+            pillow
+            pyopencl
+            pyyaml
+            requests
+            tqdm
+          ];
+
+          # TODO(b7r6): enable when all tests pass...
+          doCheck = false;
+          pythonImportsCheck = [ "tinygrad" ];
+        };
+
+        commonBuildInputs = with P; [ git nixfmt ];
+        commonPythonPackages = pypkgs:
+          with pypkgs; [
+            flake8
+            mypy
+            onnx
+            onnxruntime
+            opencv4
+            pylint
+            pytest
+            pytest-xdist
+            python-lsp-server
+            pytorch-bin
+            pyyaml
+            safetensors
+            tinygrad
+          ];
+
+        shellHook = ''
+          export __NIX_PS1__="tinygrad";
+          source "${P.bash-completion}/etc/profile.d/bash_completion.sh";
+          source "${P.git}/share/bash-completion/completions/git";
+          SCRIPT_DIR="${B.toString ./.}"
+          export PYTHONPATH="$SCRIPT_DIR/src:$PYTHONPATH"
```

Comment:
We should avoid having extra things in shellHook and can just leave it up to the user considering stuff like direnv. The PYTHONPATH should be fine to leave tho. I think the PYTHONPATH is wrong tho? shouldn't it just be `builtins.toString ./.`?

Reply:
I'll do what you want on this, I personally think that it's brittle to completely override environment variables that Nix is known to mess around with in subtle, implementation specific ways. It's generally lower risk to throw an extra entry on a `PATH` style envvar than to blow it away, but I think your proposal currently works.

LMK what you want done here.

Reply:
for the `PYTHONPATH` I meant like the `/src` at the end, shouldn't it just be the repo root?